### PR TITLE
Fix typos and formatting in TypeChecker.rst

### PR DESCRIPTION
**Pull request by sukolsak**

### CODE REVIEW

File: docs/TypeChecker.rst [hunk 1]
```diff
@@ -737,10 +737,10 @@ checking problem::
 
 This constraint system generates the constraints "``T(f)`` ==Fn ``T0
 -> T1``" (for fresh variables ``T0`` and ``T1``), "``(T2, X)`` <c
-``T0``" (for fresh variable ``T2``) and "``T2 conforms to
+``T0``" (for fresh variable ``T2``) and "``T2`` conforms to
 ``FloatLiteralConvertible``". As part of the solution, after ``T0`` is
 replaced with ``(i : Int, s : String)``, the second of
-these constraints is broken down into "``T2 <c ``Int``" and "``X`` <c
+these constraints is broken down into "``T2`` <c ``Int``" and "``X`` <c
```
Comment (benlangmuir) [inline docs/TypeChecker.rst]: The rest of the document has ``X <c Y``, rather than ``X`` <c ``Y``.

Comment (jrtibbetts) [inline docs/TypeChecker.rst]: And what about the inconsistent capitalization of `<c` vs. `<C`?

Comment (benlangmuir) [inline docs/TypeChecker.rst]: I suspect it's not worth the trouble of making it consistent, but maybe @DougGregor has an opinion.

Comment (sukolsak) [inline docs/TypeChecker.rst]: @benlangmuir Thanks. Fixed.

@jrtibbetts ``<c`` is a conversion constraint. ``<C`` is a construction constraint.

**Comment by benlangmuir** [comment]
Thanks, LGTM.

**Pull request merged**